pkg/loader/gitcloner.go

```diff
@@ -90,27 +90,42 @@ func simpleGitCloner(spec string) (
 	return checkoutDir, pathInCoDir, nil
 }
 
+func parseUrl(n string) (
+	repo string, path string, gitRef string, err error) {
+	host, repo, path, gitRef, err := parseGithubUrl(n)
+	if err != nil {
+		return
+	}
+	if isAzureHost(host) || isAWSHost(host) {
+		repo = host + repo
+		return
+	}
+	repo = host + repo + ".git"
+	return
+}
+
```

Comment:
This has no test.  Please throw in a a few, because the azure / aws thing looks very odd :)

Reply:
BTW, I'm not thrilled with how i wrote the test to create input URLS via computation, then test them
A better format would be a standard table driven test for parseUrl, e.g. something like

E.g.
```
testCases = []testData{ 
  {
    arg:            "gitHub.com/foo/bar/someDir#tag",
    expected:  {
      host: "github.com",
      repo: "foo/bar",
      path: "someDir",
      gitRef: "tag",
      err: nil ,
  },
  {
    arg:            "gh:foo/bar/someDir/subdir#tag",
    expected:  {
      host: "github.com",
      repo: "foo/bar",
      path: "someDir/subdir",
      gitRef: "tag",
      err: nil ,
  },
}
```
etc.

Reply:
I like your computed test cases, it covers so many combinations. I will modify it a little to work with the added `host`.